lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1277,9 +1277,10 @@ bool ThreadContext::IsActivePropertyId(Js::PropertyId pid)
 void ThreadContext::InvalidatePropertyRecord(const Js::PropertyRecord * propertyRecord)
 {
     InternalInvalidateProtoTypePropertyCaches(propertyRecord->GetPropertyId());     // use the internal version so we don't check for active property id
-
+#if ENABLE_NATIVE_CODEGEN
+    m_pendingJITProperties->Remove(propertyRecord);
```

Comment:
Need to update the server that the property record is removed? maybe maintain a removed list of propIds, do incremental removing similar to the incremental adding?